pallets/parachain-system/src/lib.rs

```diff
@@ -1059,13 +1062,20 @@ impl<T: Config> Pallet<T> {
 				// Thus fall through here.
 			},
 		};
-		<PendingUpwardMessages<T>>::append(message);
-		Ok(0)
+		<PendingUpwardMessages<T>>::append(message.clone());
+
+		// The relay ump does not use using_encoded
+		// We apply the same this to use the same hash
+		// TODO: shall we change both ump and this to
```

Comment:
what about this TODO?

Reply:
It seems https://github.com/paritytech/polkadot/pull/6271 will still use `sp_io::hashing::blake2_256(message)`. I guess we can leave it as is (and remove the todo), as it is the only way hashes will match